### PR TITLE
BAK-1155: Update `document` table & associated requests/responses

### PR DESCRIPTION
**Pull request by joel-wright**
- Remove user links
- add slug field
- add import_id field
- update uniqueness constraint to only include source_id & import_id

### CODE REVIEW

**Review by THOR300 — APPROVED** [review]
LGTM

File: backend/alembic/versions/0004_modify_document_table_fields.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# revision identifiers, used by Alembic.
+revision = "0004"
+down_revision = "0003"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column("document", sa.Column("slug", sa.Text(), nullable=True))
```
Comment (diversemix) [inline backend/alembic/versions/0004_modify_document_table_fields.py:21]: Don't we want slugs to be unique?

File: backend/app/db/models/geography.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ class GeoStatistics(Base):
     __tablename__ = "geo_statistics"
 
     id = sa.Column(sa.SmallInteger(), primary_key=True)
-    name = sa.Column(sa.Text(), unique=True)
+    name = sa.Column(sa.Text(), nullable=False, unique=True)
```
Comment (diversemix) [inline backend/app/db/models/geography.py:28]: I can't see this change reflected in the migrations :thinking:

**Review by diversemix — APPROVED** [review]
lgtm - a couple of points/questions

File: backend/app/db/models/geography.py (hunk 2 above)
Comment (joel-wright) [reply backend/app/db/models/geography.py:28]: I believe this is the default

File: backend/alembic/versions/0004_modify_document_table_fields.py (hunk 1 above)
Comment (joel-wright) [reply backend/alembic/versions/0004_modify_document_table_fields.py]: done

**Review by diversemix — APPROVED** [review]
Changes look good!

**Comment by joel-wright** [comment]
We finally solved the problem of the weird FK constraint. I have tested this a few times, so I think we can proceed to making this available on dev.app

**Pull request merged**